### PR TITLE
<chore>(UADSDK-1742):<Android>: Update MAX Adapter to 4.1.0

### PR DESCRIPTION
**Pull request by NicholasBar**
# Description

Update MAX Adapter to 4.1.0

[UADSDK-1742](https://jira.unity3d.com/browse/UADSDK-1742)

# How

- Updated getToken with new async API
- Updated Unity Ads SDK version to 4.1.0

# Test

- Run sample application with new adapter changes
- Updated sample application with new Unity Ads SDK version(4.1.0)
- Correctly loaded and showed ads

### CODE REVIEW

File: UnityAds/src/main/java/com/applovin/mediation/adapters/UnityAdsMediationAdapter.java [hunk 1]
```diff
@@ -146,14 +151,6 @@ public void loadInterstitialAd(final MaxAdapterResponseParameters parameters, fi
         String placementId = parameters.getThirdPartyAdPlacementId();
         log( "Loading " + ( AppLovinSdkUtils.isValidString( parameters.getBidResponse() ) ? "bidding " : "" ) + "interstitial ad for placement \"" + placementId + "\"..." );
 
-        if ( !UnityAds.isInitialized() )
```
Comment (UnitySteven) [inline UnityAds/src/main/java/com/applovin/mediation/adapters/UnityAdsMediationAdapter.java:149]: This explains a lot

**Review by UnitySteven — COMMENTED** [review]
Before initialize can you set the following metadata:
// Set mediation meta data before initializing.
    MediationMetaData mediationMetaData = new MediationMetaData(context);
    mediationMetaData.setName("Max");
    mediationMetaData.setVersion(UnityAds.getVersion());
    mediationMetaData.set("adapter_version", libraryVersionCode);
    mediationMetaData.commit();

File: UnityAds/src/main/java/com/applovin/mediation/adapters/UnityAdsMediationAdapter.java (hunk 1 above)
Comment (NicholasBar) [reply UnityAds/src/main/java/com/applovin/mediation/adapters/UnityAdsMediationAdapter.java:149]: Do we want to remove these for all types of Ads?

**Comment by NicholasBar** [comment]
@UnitySteven not saw if you saw line 77
It doesn't enter that block for me, also seems format might be different from what you mentioned in your comment.

setName resolves to 
```
protected static String mediationTag() {
        return "AppLovinSdk_" + AppLovinSdk.VERSION;
    }
```

File: UnityAds/src/main/java/com/applovin/mediation/adapters/UnityAdsMediationAdapter.java (hunk 1 above)
Comment (UnitySteven) [reply UnityAds/src/main/java/com/applovin/mediation/adapters/UnityAdsMediationAdapter.java:149]: yes

File: UnityAds/build.gradle.kts [hunk 2]
```diff
@@ -4,9 +4,9 @@ plugins {
 }
 
 private val versionMajor = 4
-private val versionMinor = 0
-private val versionPatch = 1
-private val versionAdapterPatch = 2
+private val versionMinor = 1
+private val versionPatch = 0
+private val versionAdapterPatch = 3
```
Comment (lmoreault) [inline UnityAds/build.gradle.kts:9]: Should the adapter patch go back down to 0? (I think so, so the adapter version is 4.1.0.0)

**Review by lmoreault — APPROVED** [review]
Aside from `versionAdapterPatch` I think the rest looks good to me :)

**Comment by UnitySteven** [comment]
close?

**Comment by NicholasBar** [comment]
Closing and fetching directly from MAXs update

**Comment by NicholasBar** [comment]
Closing and updating/pulling from MAXs actually PR

**Pull request closed without merging**